src/WebApi/Application/UseCases/QuestionSet/DeleteQuestionSet/DeleteQuestionSetUseCase.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Application.Repositories;
+using System.Threading.Tasks;
+
+namespace Application.UseCases.QuestionSet.DeleteQuestionSet
+{
+    public class DeleteQuestionSetUseCase : IDeleteQuestionSetUseCase
+    {
+
+        private IOutputPort _outputPort;
+
+        private readonly IQuestionSetRepository _questionSetRepository;
+
+        public DeleteQuestionSetUseCase(IQuestionSetRepository questionSetRepository)
+        {
+            _questionSetRepository = questionSetRepository;
+        }
+
+        public async Task Execute(DeleteQuestionSetInput input)
+        {
+            var questionSet = await _questionSetRepository.GetById(input.Id);
+
+            if (questionSet is null)
+            {
+                _outputPort.NotFound();
+                return;
+            }
+
+            bool isDeleted = await _questionSetRepository.Delete(input.Id);
```

Comment:
It may be too unimportant a detail but we don't need to define type on the right side since it's defined by the return value.  Maybe I am missing something here, but there is a suggestion to consider.
```suggestion
            var isDeleted = await _questionSetRepository.Delete(input.Id);
```